packages/device_info/lib/device_info.dart

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
```

Comment:
Crash reporting done within Dart VM does not seem very useful to me. Flutter apps can crash on native side as well which means you will need two ways of doing crash reporting.

Why don't we write a plugin for Firebase Crash instead? Works both on iOS and Android and you can log custom crash logs with it. AFAIK, all device information is automatically collected.

I can see some edge cases where inspecting device information on Flutter might be useful but I believe these are rare edge cases so I would not nitpick too much about the proper pattern for this plugin. There's a reason why it is not a good fit.